CSS property update: counter-increment (#29688)
mdn/content: files/en-us/web/css/counter-increment/index.md

{{CSSRef}}

> The **`counter-increment`** [CSS](/en-US/docs/Web/CSS) property increases or decreases the value of the named [CSS counter](/en-US/docs/Web/CSS/CSS_counter_styles/Using_CSS_counters) by the specified value or increments it by `1` if no value is provided.

Reviewer: Suggesting to change the position of the "counter-reset" sentence. Currently, it is lost between the interactive example and "Syntax" sections.

```suggestion
The **`counter-increment`** [CSS](/en-US/docs/Web/CSS) property increases or decreases the value of the named [CSS counter](/en-US/docs/Web/CSS/CSS_counter_styles/Using_CSS_counters) by the specified value or increments it by `1` if no value is provided.

You can reset the counter's value to any integer value by using the {{cssxref("counter-reset")}} CSS property.
```

{{EmbedInteractiveExample("pages/css/counter-increment.html")}}

> You can reset the counter's value to any integer value by using the {{cssxref("counter-reset")}} CSS property.

Reviewer: Moving it before the interactive example
```suggestion

```

counter-increment: unset;
```

### Values

> The `counter-increment` property takes as its value either a list of space-separated counter names specified as `<custom-ident>` with an optional `<integer>` value or the keyword `none`.

Reviewer: How about moving the second sentence up before the description of all values?
```suggestion
The `counter-increment` property takes as its value either a list of space-separated counter names specified as `<custom-ident>` with an optional `<integer>` value or the keyword `none`. You may specify as many counters to increment as you want, with each name or name-number pair separated by a space.
```

- {{cssxref("&lt;custom-ident&gt;")}}
  - : Specifies the name of the counter to increase or decrease.
- {{cssxref("&lt;integer&gt;")}}
  - : Specifies the value to add to the counter. If the integer is preceded by a `-` sign, the value will be subtracted from the counter. Defaults to `1` if no value is specified.
- `none`
  - : Indicates that no counter must be increased or decreased. This value can also be used to cancel all counters from being increased or decreased in more specific rules. This is the default value of the property.

> 
> You may specify as many counters to increment as you want, with each name or name-number pair separated by a space.

Reviewer: ```suggestion
```

> > **Note:** Using the `none` value prevents all counters from being increased or decreased for the selected elements where this rule applies. To prevent only specific counters from being increased or decreased, set the value as `0` on the relevant counter(s).

Reviewer: This note might need another indent to line up under `none`.
```suggestion
>> **Note:** Using the `none` value prevents all counters from being increased or decreased for the selected elements where this rule applies. To prevent only specific counters from being increased or decreased, set the `integer` value as `0` on the relevant counter(s).
```

### Values

The `counter-increment` property takes as its value a list of space separated list of counter names `<custom-ident>` with an optional `<integer>` incrementor or signed decrementor, or the keyword `none`.

> 

Author: removed the lines. added " You may specify as many counters to increment as you want, with each name or name-number pair separated by a space." back in after the values in the values section but before the note that is there

## Examples

### Decreasing the counter value

In this example, we display a sequence of numbers counting backward. To do this, we use a counter to display numbers starting from 100 and decreasing by 7 each time.

#### HTML

```html
<div>
  <i></i><i></i><i></i><i></i><i></i><i></i><i></i> <i></i><i></i><i></i><i></i
  ><i></i><i></i><i></i> <i></i><i></i><i></i><i></i><i></i><i></i><i></i>
  <i></i><i></i><i></i><i></i><i></i><i></i><i></i>
</div>
```

#### CSS

We set the initial value of the counter named `sevens` to `100` by using {{cssxref("counter-reset")}}. Then, for each {{HTMLElement("i")}}, we decrease the counter by `7`.

> To set the first count at `100`, we target the first `<i>` element by using the {{cssxref(":first-of-type")}} pseudo-class and setting `counter-increment: none;`. Additionally, the {{cssxref("content")}} property is used in the {{cssxref("::before")}} pseudo-element to display the value of the counter using the [`counter()`](/en-US/docs/Web/CSS/counter) function."

Reviewer: ```suggestion
To set the first count at `100`, we target the first `<i>` element by using the {{cssxref(":first-of-type")}} pseudo-class and setting `counter-increment: none;`. Additionally, the {{cssxref("content")}} property is used in the {{cssxref("::before")}} pseudo-element to display the value of the counter using the [`counter()`](/en-US/docs/Web/CSS/counter) function.
```

## Examples

### Decrementing

In this example, we count backward by 7 starting with 100.

#### HTML

```html
<div>
  <i></i><i></i><i></i><i></i><i></i><i></i><i></i> <i></i><i></i><i></i><i></i
  ><i></i><i></i><i></i> <i></i><i></i><i></i><i></i><i></i><i></i><i></i>
  <i></i><i></i><i></i><i></i><i></i><i></i><i></i>
</div>
```

#### CSS

> We set the `sevens` counter to `100` using {{cssxref("counter-reset")}} then decrement the counter by `7` every for every {{HTMLElement("i")}}.
> 
> As we want the first count to be `100`, using the {{cssxref(":first-of-type")}} pseudoclass we prevent the counter from decrementing the first `<i>` by setting `counter-increment: none;`.

Reviewer: We'll need to say something about `content` property and `counter()` function.
```suggestion
We set the initial value of the counter named `sevens` to `100` by using {{cssxref("counter-reset")}}. Then, for each {{HTMLElement("i")}}, we decrease the counter by `7`.

To set the first count at `100`, we target the first `<i>` element by using the {{cssxref(":first-of-type")}} pseudo-class and setting `counter-increment: none;`. Additionally, the {{cssxref("content")}} property is used in the {{cssxref("::before")}} pseudo-element to display the value of the counter using the [`counter()`](/en-US/docs/Web/CSS/counter) function."
```

```css
div {
  counter-reset: sevens 100;
}
i {
  counter-increment: sevens -7;
}
i:first-of-type {
  counter-increment: none;
}
i::before {
  content: counter(sevens);
}
```

```css hidden
div {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: 300px;
  width: 200px;
}
i {
  flex: 0 0 2em;
}
```

#### Result

> {{EmbedLiveSample("Decrementing", 140, 300)}}

Reviewer: ```suggestion
{{EmbedLiveSample("Decreasing the counter value", 140, 300)}}
```

## See also

> - {{cssxref("counter-reset")}}
> - {{cssxref("counter-set")}}
> - [Using CSS counters](/en-US/docs/Web/CSS/CSS_counter_styles/Using_CSS_counters)
> - {{cssxref("@counter-style")}}
> - The {{cssxref("counter", "counter()")}} and {{cssxref("counters", "counters()")}} functions
> - [CSS lists and counters](/en-US/docs/Web/CSS/CSS_lists) module
> - [CSS counter styles](/en-US/docs/Web/CSS/CSS_counter_styles) module

Reviewer: How about grouping like this:
```suggestion
- Counter properties: {{cssxref("counter-set")}}, {{cssxref("counter-reset")}}
- Counter at-rule: {{cssxref("@counter-style")}}
- Counter functions: {{cssxref("counter", "counter()")}}, {{cssxref("counters", "counters()")}}
- [Using CSS counters](/en-US/docs/Web/CSS/CSS_counter_styles/Using_CSS_counters) guide
- [CSS lists](/en-US/docs/Web/CSS/CSS_lists) module
- [CSS counter styles](/en-US/docs/Web/CSS/CSS_counter_styles) module
```
